data_models/aws_cloudtrail_data_model.yml

```diff
@@ -0,0 +1,15 @@
+AnalysisType: datamodel
+LogTypes: 
+  - AWS.CloudTrail
+DataModelID: AWS.CloudTrail.DataModel
+Filename: aws_cloudtrail_data_model.py
+Enabled: true
+Mappings:
+  - Name: actor_user
+    Path: $.userIdentity.userName
```

Comment:
Is there some fancy regex pathing we could do here to also catch `$.userIdentity.sessionContext.sessionIssuer.userName` here? In the case of an assumed role that's where you'd actually find the `actor_user` involved.

Reply:
Is this a JSONPath expression? If so:
`$.userIdentity..userName`

(there might be a way to scope it more precisely)

- https://jsonpath2.readthedocs.io/en/latest/exampleusage.html#examples
- https://jsonpath.com/

Reply:
It looks like there is a bug in the "union" operator for `jsonpath_ng`, I'll use this recursive expression, thanks!